server/internal/server/server.go

```diff
@@ -154,7 +150,33 @@ func (s *S) organizationRole(orgID, userID string) v1.OrganizationRole {
 	return v1.OrganizationRole(r)
 }
 
-func (s *S) validateOrganizationOwner(orgID, userID string) error {
+func (s *S) canReadOrganization(orgID, userID string) bool {
+	if !s.enableAuth {
+		return true
+	}
+
+	return s.organizationRole(orgID, userID) != v1.OrganizationRole_ORGANIZATION_ROLE_UNSPECIFIED
+}
+
+func (s *S) canManageOrganization(orgID, userID string) bool {
+	if !s.enableAuth {
+		return true
+	}
+
+	return s.organizationRole(orgID, userID) == v1.OrganizationRole_ORGANIZATION_ROLE_OWNER
+}
+
+// validateUserForManagingOrg checks if the user has the permission to manage the organization.
+//
+// If the authorization is enabled, this passes only when the user is an owner of the organization.
+//
+// If the authorization is disabled, this passes.
+// Note that in that case, the existence of the organization is not checked.
```

Comment:
added this note to make sure we call `validateOrganizationID()` around this method call when we use this at some new place in the future.